src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -434,6 +447,14 @@ private static function createRedirectResolver(array $options, string $host, int
         };
     }
 
+    private function ensureState(): CurlClientState
+    {
+        $this->multi ??= new CurlClientState($this->maxHostConnections, $this->maxPendingPushes);
+        $this->multi->logger = $this->logger;
```

Comment:
should this really be done each time `ensureState` is called or only when instantiating the state ?

Reply:
No, logger only needs to be set on construction, any other time it changes will pass via the isset-test in setLogger.

The effect will be the same, but it is a minor papercut that can be removed. I pushed a commit to fix that.